keys-component.js

```diff
@@ -59,7 +66,11 @@ function factory($scope, $routeParams, brAlertService, brKeyService,
     self.modals.key = null;
   };
 
-  self.init = function(identity) {
+  self.onKeyGenerate = function(key) {
```

Comment:
Tried a few more things to address this with no success.  However, found that you can make the generate key modal close with a mouse-over on the key action menu (upper right).